DirectEmployers/MyJobs
# Datadog for celery.

## What's this PR do?
Sets up Datadog for celery. 
Removes a bunch of stuff that is irrelevant if Datadog becomes a thing, and needs replaced anyway if it doesn't.
I also have a better understanding of how traces and the AGENT_HOSTNAME setting works, so remove the middleware and use that instead.
Because of the above point, I'm going to take another stab at getting Jenkins working with APM.

## Where should the reviewer start?
Major changes are in imports. 
There should be no actual change to how the code functions.

## How should this be manually tested?
A very, very limited version of the imports changes are already in production, so check out Datadog for that.
For the local changes, in settings.py set `DATADOG_TRACE['ENABLED'] = True`. Please also change the env tag to something specific to you (I've been using my name).


## Any background context you want to provide?
@darrint asked me to do some voodoo with the decorators, so we can hide a lot of the logic and make it less Datadog-y (like I tried to do with the `log_dict` function). 
This is messy and impractical at the moment, especially when, if we don't go with Datadog, the answer is going to be to pull it all out.



### Separation of Concerns
